File: prop-archaius/src/main/java/io/pleo/prop/archaius/ArchaiusPropFactory.java
package io.pleo.prop.archaius;

import java.util.function.Function;

import io.pleo.prop.core.Prop;
import io.pleo.prop.core.internal.PropFactory;

public class ArchaiusPropFactory implements PropFactory {

  @Override
  public <T> Prop<T> createProp(String propName, Function<String, T> parse, T defaultValue) {
    return new ArchaiusProp<>(new ParsingProperty<>(propName, parse, defaultValue));
  }
}

File: prop-archaius/src/main/kotlin/io/pleo/prop/archaius/ArchaiusPropFactory.kt
package io.pleo.prop.archaius

import io.pleo.prop.core.Parser
import io.pleo.prop.core.Prop
import io.pleo.prop.core.internal.PropFactory

class ArchaiusPropFactory : PropFactory {
    override fun <T> createProp(
        propName: String,
        parse: Parser<T>,
        defaultValue: T?,
    ): Prop<T> =
        ArchaiusProp(
            ParsingProperty(propName, parse, defaultValue),
        )
}
